src/halmos/bytevec.py

```diff
@@ -142,6 +142,33 @@ def __getitem__(self, key) -> Byte | ForwardRef("Chunk"):
     def __len__(self) -> int:
         return self.length
 
+    def concretize(self, substitution: dict[BitVecRef, BitVecRef]) -> "Chunk":
+        """
+        Replace all symbols in the data with their corresponding concrete values,
+        if they exist in the given substitution mapping.
+
+        Return a new object with concrete values, or self if no substitution is made.
+        """
+
+        if not isinstance(self, SymbolicChunk):
+            return self
+
+        data = self.data
+        # TODO: call substitute() only when the data contains variables to be substituted
+        new_data = substitute(data, *substitution.items())
```

Comment:
I am slightly worried about the perf impact of `substitute` always constructing a new object, but I guess we'll see how it looks like when I profile (not a blocker)

Reply:
yeah, proper profiling would be greatly helpful. based on my performance tests, it gets slower if data is a complex term, e.g., LibUint1024Test:testProveSubAdd().

i'm thinking applying this substitution to selected `int_of()` only, rather than blindly applying to all calldatacopy. would be a separate pr though.